plumbing/object/commit.go

```diff
@@ -341,6 +345,12 @@ func (c *Commit) encode(o plumbing.EncodedObject, includeSig bool) (err error) {
 		}
 	}
 
+	for _, header := range c.ExtraHeaders {
+		if _, err = fmt.Fprintf(w, "\n%s", header); err != nil {
+			return err
+		}
+	}
+
```

Comment:
Please add some validation around this logic. I'm not completely sure of all the validation that takes place upstream, but it would be good to explore adding checks for:
- Empty values
- Headers with the same name as core headers (`tree`, `author`, `committer`, `parent`, etc.)
- Header names without values
- Header values without names

Whatever we can confirm upstream blocks, we should also block.

Reply:
Does this version works for you?